universe/base.go

```diff
@@ -207,10 +219,11 @@ func (a *MintingArchive) RegisterIssuance(ctx context.Context, id Identifier,
 // verifyIssuanceProof verifies the passed minting leaf is a valid issuance
 // proof, returning the asset snapshot if so.
 func (a *MintingArchive) verifyIssuanceProof(ctx context.Context, id Identifier,
-	key BaseKey, leaf *MintingLeaf) (*proof.AssetSnapshot, error) {
+	key BaseKey, leaf *MintingLeaf,
+	prevAssetSnapshot *proof.AssetSnapshot) (*proof.AssetSnapshot, error) {
 
 	assetSnapshot, err := leaf.GenesisProof.Verify(
-		ctx, nil, a.cfg.HeaderVerifier,
+		ctx, prevAssetSnapshot, a.cfg.HeaderVerifier,
```

Comment:
I guess we can also rename `GenesisProof` above?

Reply:
Yes, there's also a `BaseKey` and a few other things to rename. I've got a separate PR on the go for that.